gtsfm/evaluation/metrics.py

```diff
@@ -0,0 +1,399 @@
+"""Classes to store metrics computed in different GTSfM modules.
+
+The GtsfmMetric class stores a single metric, and the GtsfmMetricsGroup stores a list of metrics.
+These classes are used to compute statistics on the metrics (min, max, etc), 
+save them to JSON, parse metrics from JSON, and plot them.
+
+Authors: Akshay Krishnan
+"""
+
+from __future__ import annotations
```

Comment:
any particular reason to include these? i think past 3.5.0 type hints are included, and 3.8 is our supported platform

Reply:
we are using the class name as a type hint for methods in the class itself. 
https://stackoverflow.com/questions/33533148/how-do-i-type-hint-a-method-with-the-type-of-the-enclosing-class